### ISSUE TITLE
Testing

### ISSUE THREAD

**Issue by soyersoyer** [soyersoyer/sysex-controls#2]
I don't have access to all supported devices. Please share your experiences. What works, what doesn't, what could be better.

**Comment by Kris33** [comment]
Sysex-controls also work on KDE out of the box. 
Arturia Keylab Essential 88 MKI not supported though.

**Comment by rdbende** [comment]
The MiniLab mkII also works well. My only "issue" is that the UI is a bit hard to parse. It takes time to find the control I'm looking for, and even when I find it, I have to read through the rows individually to figure out how it is configured. It would be great if the widgets would be a bit more graphical.
I had a similar project in the works, but it has been stagnating, and since this project looks much more polished and supports more models, I thought I'd contribute some of my ideas to make the UI less "text-based". I think I'll drop a PR at the weekend, as I already made a few developments, but left them in a local repo at home.

**Comment by soyersoyer** [comment]
> Arturia Keylab Essential 88 MKI not supported though.

I've just added it to the main branch. If you can, please try it.

**Comment by soyersoyer** [comment]
> The MiniLab mkII also works well. My only "issue" is that the UI is a bit hard to parse. It takes time to find the control I'm looking for, and even when I find it, I have to read through the rows individually to figure out how it is configured. It would be great if the widgets would be a bit more graphical. I had a similar project in the works, but it has been stagnating, and since this project looks much more polished and supports more models, I thought I'd contribute some of my ideas to make the UI less "text-based". I think I'll drop a PR at the weekend, as I already made a few developments, but left them in a local repo at home.

Sounds good. I look forward to your changes! :)

**Comment by Kris33** [comment]
> > Arturia Keylab Essential 88 MKI not supported though.
> 
> I've just added it to the main branch. If you can, please try it.

well I just got the upgraded flatpack. but wether I select Arturia Keylab Essential DAW or MID, I get the "control value read failed " error message.

**Comment by soyersoyer** [comment]
Can you run it from console and copy the output?
`flatpak run hu.irl.sysex-controls`

**Comment by Kris33** [comment]
> Can you run it from console and copy the output?
> 
Of course, here you go.

> flatpak run hu.irl.sysex-controls
> ```
(sysex-controls:2): Gdk-WARNING **: 08:50:43.626: ../gdk/wayland/gdkcursor-wayland.c:210 cursor image size (32) not an integer multiple of theme size (24)

(sysex-controls:2): Gdk-WARNING **: 08:50:43.635: ../gdk/wayland/gdkcursor-wayland.c:210 cursor image size (32) not an integer multiple of theme size (24)

(sysex-controls:2): Gdk-WARNING **: 08:50:43.685: ../gdk/wayland/gdkcursor-wayland.c:210 cursor image size (32) not an integer multiple of theme size (24)
sc_midi_arturia_read_next(02, 00004006) poll timeout 0

** (sysex-controls:2): WARNING **: 08:50:47.558: control value read failed
sc_midi_arturia_read_next(02, 00000230) poll timeout 0

** (sysex-controls:2): WARNING **: 08:50:52.581: control value read failed

**Comment by soyersoyer** [comment]
It is interesting that it stopped somewhere else at the second opening.
Could you please create a dump from the Midi Control Center?
- Start the MIDI Control Center
- Enable MIDI Console in View/Midi Console.
- Connect your controller
- Copy the Sysex values from the console

**Comment by Kris33** [comment]
> It is interesting that it stopped somewhere else at the second opening. Could you please create a dump from the Midi Control Center?
> 
>     * Start the MIDI Control Center
> 
>     * Enable MIDI Console in View/Midi Console.
> 
>     * Connect your controller
> 
>     * Copy the Sysex values from the console

Sadly I can't. MIDI Control Center doesn't run well on fedora, hangs after first click on wine.
I don't have a windows machine to test.

**Comment by soyersoyer** [comment]
MCC also works on a virtual machine. I can't figure out the bug without your dump.

**Comment by plamikcho** [comment]
Does it work on MacOS (Apple silicon)? I tried to build it but ended with complains about missing dependencies, installed as much as possible using homebrew but one of them - 'alsa' is not available.

**Comment by soyersoyer** [comment]
The concept of Apple privacy is increasingly different from mine, it is unlikely that I will have an Apple machine in the future, so I cannot support MacOS, sorry.

**Comment by soyersoyer** [comment]
@Kris33 The v0.2.17 has a fix for timeout issues. Could you try it?

**Comment by soyersoyer** [comment]
This issue is too broad, if something doesn't work please create a controller specific issue.
